src/gui/shareusergroupwidget.cpp

```diff
@@ -650,24 +650,30 @@ void ShareUserLine::loadAvatar()
      *
      * This will be shown until the avatar image data arrives.
      */
-    const QByteArray hash = QCryptographicHash::hash(_ui->sharedWith->text().toUtf8(), QCryptographicHash::Md5);
-    double hue = static_cast<quint8>(hash[0]) / 255.;
 
     // See core/js/placeholder.js for details on colors and styling
-    const QColor bg = QColor::fromHslF(hue, 0.7, 0.68);
+    const auto backgroundColor = backgroundColorForShareeType(_share->getShareWith()->type());
     const QString style = QString(R"(* {
         color: #fff;
         background-color: %1;
         border-radius: %2px;
         text-align: center;
         line-height: %2px;
         font-size: %2px;
-    })").arg(bg.name(), QString::number(avatarSize / 2));
+    })").arg(backgroundColor.name(), QString::number(avatarSize / 2));
     _ui->avatar->setStyleSheet(style);
 
     // The avatar label is the first character of the user name.
     const QString text = _share->getShareWith()->displayName();
-    _ui->avatar->setText(text.at(0).toUpper());
+
+    const auto pixmap = pixmapForShareeType(_share->getShareWith()->type(), backgroundColor);
+
+    if (!pixmap.isNull()) {
+        _ui->avatar->setPixmap(pixmap);
+    } else {
+        qCDebug(lcSharing) << "pixmap is null for share type: " << _share->getShareWith()->type();
+        _ui->avatar->setText(text.at(0).toUpper());
+    }
```

Comment:
Maybe you could factor that out into a separate function called `setAvatarPixmap` or something more fitting :)